src/js/login/components/SearchHelpSignIn.jsx

```diff
@@ -12,12 +12,15 @@ import SignInProfileMenu from './SignInProfileMenu';
 import { toggleLoginModal, toggleSearchHelpUserMenu } from '../actions';
 import { isUserRegisteredForBeta } from '../../personalization/beta-enrollment/actions';
 
-class SearchHelpSignIn extends React.Component {
-  componentDidMount() {
-    const nextParams = new URLSearchParams(window.location.search);
-    const nextPath = nextParams.get('next');
-    if (nextPath) {
-      this.props.toggleLoginModal(true);
+export class SearchHelpSignIn extends React.Component {
+  componentDidUpdate() {
```

Comment:
Changed to `componentDidUpdate` part of the lifecycle since `currentlyLoggedIn` only gets determined after the initial user data fetch, and we need to confirm that before we redirect for login.

Reply:
Going to make a correction to the conditional; the way it is currently, it won't close the modal if you try to X out since it's updating the state, but `showModal` would be false. Needs to be sure that this only happens when the modal has remained closed.